### PR TITLE
Load Modules Needed for Building on Summit

### PR DESCRIPTION
**Pull request by ben-e-whitney**
I encountered a few missing modules while testing the build scripts on Summit.

### CODE REVIEW

File: build_scripts/build_mgard_cuda_turing.sh [hunk 1]
```diff
@@ -9,6 +9,9 @@
 set -e
 set -x
 
+module load cuda/11.4
+module load cmake
+
```
Comment (JieyangChen7) [inline build_scripts/build_mgard_cuda_turing.sh:14]: @ben-e-whitney This build script is for NVIDIA Turing GPU architectures (e.g., the sysytem76 machine in Scott's group) not for Volta GPUs on Summit. So there is no need to load modules.
For Summit, I have provided a dedicated build script `build_mgard_cuda_summit.sh`.

Comment (ben-e-whitney) [reply build_scripts/build_mgard_cuda_turing.sh:14]: Oh, OK. My mistake! I was trying to build everything. I'll revert this change. Should I keep the change to `build_scripts/build_mgard_serial.sh`?

File: build_scripts/build_mgard_serial.sh [hunk 2]
```diff
@@ -9,6 +9,8 @@
 set -e
 set -x
 
+module load cmake
+
```
Comment (JieyangChen7) [inline build_scripts/build_mgard_serial.sh:13]: The Summit build script `build_mgard_cuda_summit.sh` already builds both the CUDA and Serial backend in MGARD-X. This `build_mgard_serial.sh` is solely for building just the Serial backend on an arbitrary machine (e.g., any machine with only CPUs), so I'm not sure it is a good idea to have `module load` here.

Comment (ben-e-whitney) [reply build_scripts/build_mgard_serial.sh:13]: Oh, I didn't notice that `build_mgard_cuda_summit.sh` builds the serial backend. I will just delete this whole pull request, then. Thank you for reviewing, @JieyangChen7!

File: build_scripts/build_mgard_cuda_turing.sh (hunk 1 above)
Comment (JieyangChen7) [reply build_scripts/build_mgard_cuda_turing.sh:14]: I think `build_mgard_cuda_summit.sh` is enough to build everything on Summit, so it is better to keep `build_mgard_serial.sh` unchanged for now.

**Pull request closed without merging**